main.go

```diff
@@ -120,7 +120,7 @@ func main() {
 
 	metricsAPI := metrics.New()
 	metricsAPI.Register()
-	registryAPI := registry.NewRegistry()
+	registryAPI := registry.NewRegistry(client)
```

Comment:
I would really prefer not to wire those two.
Does it have an additional benefit rather than shorten the code? I also feel it is making the code less comprehensive.